src/com/madalinb/coffeeshop/csv/CSVParser.kt

```diff
@@ -0,0 +1,31 @@
+package com.madalinb.coffeeshop.csv
+
+import com.madalinb.coffeeshop.abstractions.Parser
+import com.madalinb.coffeeshop.abstractions.Validator
+import com.madalinb.coffeeshop.data.CoffeeShop
+
+class CSVParser(var validator: Validator) :
+    Parser<CoffeeShop> {
+    override fun parseLine(line: String): CoffeeShop? {
+        val params = line.split(",")
+
+        if (validator.hasEnoughArguments(params)) {
+            if (validator.isValidName(params[0])) {
+                if (validator.isValidDouble(params[1]) &&
+                    validator.isValidDouble(params[2])
+                ) {
+                    return CoffeeShop(
```

Comment:
This reminds me of https://www.audero.it/blog/wp-content/uploads/2016/08/php-street-fighter.jpg

I think this validator has a very large api surface, it should simply state if a line is valid or not.

Can you think of a better way to do this?

Reply:
Refactored the class, and introduced Exceptions.